src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -89,6 +89,12 @@ public OpenIdConnectHandler(HttpClient backchannel)
                     message.PostLogoutRedirectUri = Options.PostLogoutRedirectUri;
                 }
 
+                var ticket = await HandleAuthenticateOnceAsync();
+                if (ticket != null)
+                {
+                    message.IdTokenHint = ticket.Principal?.FindFirst(OpenIdConnectParameterNames.IdToken)?.Value;
```

Comment:
Doesn't this have a pretty strong assumption how the original id token was preserved.  Isn't the Katana middleware using the bootstrap context for the same purpose?